### ISSUE TITLE
[Bug] Positional-only Parameters Are Dead Broke

### ISSUE THREAD

**Issue by dycw** [beartype/beartype#101]
Hi, been looking forward to trying out this new release!

Unfortunately, right out-of-the-box I think `Union`s are giving me a problem:

```python

from typing import Any
from typing import Union

from beartype import beartype

@beartype                              
def func(x: Union[int, str], /) -> Any:
    return x
```

This is giving me:

```
src/tests/test_foo.py:12: in <module>
    func(1)
<@beartype(tests.test_foo.func) at 0x7f2523357a60>:9: in func
    ???
E   NameError: name '__beartype_args_len' is not defined
```

Currently I'm using

```
> python
Python 3.9.10 (main, Feb 11 2022, 15:23:19)
[GCC 9.3.1 20200408 (Red Hat 9.3.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
```

**Comment by leycec** [comment]
`</generous_facepalm>`

I have frightening Friday-night news and I have less frightening Friday-night news. Let's start with the latter, because I already suspect this wilderness snow-bound cottage is haunted by dead nuns. :ghost: 

## What Yet Lives Is Good

Unions are still alive and kicking righteous donkey in `beartype 0.10.0`, for which we all give thanks:

```python
from typing import Any
from typing import Union

from beartype import beartype

# PEP 484-style unions are still alive.
@beartype                              
def func(x: Union[int, str]) -> Any:
    return x

# PEP 604-style unions are now alive, too.
@beartype                              
def hotter_func(x: int | str) -> Any:
    return x
```

## What Lies Bleeding Out Everywhere Is Bad

What is *not* alive and kicking righteous donkey in `beartype 0.10.0` are positional-only parameters, which `beartype 0.10.0` newly introduced but failed to adequately test. We now curse the slovenly bear for hibernating in its den when it should have been furiously mashing on a keyboard.

Since this is a critical issue, I will **resolve this immediately.** If immediately isn't fast enough (*...sometimes it's just not*), you can temporarily hack around this on your end by dropping the positional-only requirement (i.e., removing the `, /` syntax). I acknowledge that's unacceptable in most cases, so... *yeah.* It's an awkward moment we're having here.

The only silver lining to this ominous thundercloud is that `beartype < 0.10.0` didn't even support positional-only parameters. Not that this is much better, mind you... The exhaustion we feel is real. :face_exhaling:

**Comment by leycec** [comment]
**Resolved by d42bb5a.** Because this is critical breakage, I'm immediately gestating a new stable release. Let us pray it actually is.

**Incoming:** `beartype 0.10.1`. :bomb: :boom:

**Comment by leycec** [comment]
**Released in `beartype 0.10.1`.** Let us know if that returns sanity to your typing workflow! :hot_face: 

```bash
$ pip install --upgrade beartype
```
